src/libcoretest/iter.rs

```diff
@@ -916,6 +916,20 @@ fn test_range_step() {
 }
 
 #[test]
+#[should_panic]
+fn test_range_overflow_unsigned() {
+    let mut it = u8::MAX..;
```

Comment:
Ahh, that's a good point. Then these have to be run-fail with the overflow checking forced on (see `src/test/run-fail/*overflow*.rs`).